### ISSUE TITLE
[prefer-readonly-type] doesn't catch mutable records

### ISSUE THREAD

**Issue by danielnixon** [eslint-functional/eslint-plugin-functional#150]
`prefer-readonly-type` catches this:

```ts
type Foo = {
  [K in string]: string;
};
```

but not this, which is equivalent:

```ts
type Bar = Record<string, string>;
```

It would be great if `prefer-readonly-type` could be extended to catch `Record` (but _not_ `Readonly<Record<...>>`), and even better if it had a fixer that replaced the above with:

```ts
type Bar = Readonly<Record<string, string>>;
```

**Comment by RebeccaStevens** [comment]
Unfortunately we don't support `Readonly<T>` at all yet - support for it being tracked in #51 

I believe this issue is will be resolved once support is added.

**Comment by danielnixon** [comment]
Even without `Readonly<T>` support we could still flag `Record` as mutable? Or would that be too obnoxious because the obvious fix (`Readonly<Record<...>>`) wouldn't appease the linter?

**Comment by RebeccaStevens** [comment]
Yeah, if we mark `Record<T>` as a mutable type, you would get an error trying to use it in `Readonly<Record<T>>`. #51 needs to be addressed first so that mutable types wrapped by `Readonly` aren't flagged as errors.

**Comment by danielnixon** [comment]
Thanks Rebecca. My hack to flag `Record` for now in case anyone else lands here before #51 is this: https://github.com/danielnixon/eslint-config-typed-fp/blob/6c989552c16a12d86e300659686a63c42637ab18/src/index.ts#L66-L75
